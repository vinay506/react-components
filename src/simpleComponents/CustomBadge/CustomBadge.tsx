import React from 'react'
import styled from 'styled-components';
function CustomBadge(props: any) {
    /**
     * Text: Content to be displayed
     * bgColor: Backhround color of a badge
     * color: color of the text
     * position: Use when we want to display the batch over any container. Make sure make your container position as relative
     * (top, right, left, bottom): If position is absolute then use these properties to display the bacth over any container at any position
     * borderRadius: If required but default is 5px
     */
    const { text, bgColor, color, position, top, right, left, bottom, borderRadius } = props;
    const badgeInlineStyle = {
        backgroundColor: bgColor,
        color: color,
        position: position,
        top: top,
        right: right,
        left: left,
        bottom: bottom,
        borderRadius: borderRadius
    };
    return (
        <React.Fragment>
            <BadgeSpan id="uniqueBadge" style={badgeInlineStyle}>
                {text}
            </BadgeSpan>
        </React.Fragment>
    )
}

const BadgeSpan = styled.span`
    background-color: blue;
    color: white;
    padding: 5px;
    border-radius: 5px;
    height: 10px;
    min-width: 20px;
    font-size: 12px;
`

export default CustomBadge
