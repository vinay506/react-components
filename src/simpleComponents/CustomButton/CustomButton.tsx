import React from 'react'
import { Button } from '@rmwc/button';
import '@rmwc/button/styles';
export const BUTTON_TYPES = {
    RAISED: 'RAISED',
    UNELEVATED: 'UNELEVATED',
    OUTLINED: 'OUTLINED',
    DENSE: 'DENSE'
}
function CustomButton(props: any) {
    /**
     * Text: The content to be displayed on the button
     * type: Constants provided in BUTTON_TYPES can be used
     * onButtonClick: is a call back function when user clicks on the button
     * color: Text color on the button
     * bgColor: Background color of the button
     * borderColor: Border color of the button 
     */
    const { text, type, onButtonClick, color, bgColor, borderColor,width,height,padding,margin } = props;
    const buttonStyle = {
        color: color,
        backgroundColor: bgColor,
        borderColor: borderColor,
        width:width,
        height:height,
        padding:padding,
        margin:margin
    };

    let button;
    switch (type) {
        case BUTTON_TYPES.RAISED:
            button = <><Button id="raised" raised label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break;
        case BUTTON_TYPES.DENSE:
            button = <><Button id="dense" dense label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break;
        case BUTTON_TYPES.OUTLINED:
            button = <><Button id="outlined" outlined label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break;
        case BUTTON_TYPES.UNELEVATED:
            button = <><Button id="unelevated" unelevated label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break
        default:
            button = <><Button raised label={text} style={buttonStyle} onClick={onButtonClick}></Button></>
            break;
    }
    return (
        <React.Fragment>
            {button}
        </React.Fragment>
    )
}

export default CustomButton;


