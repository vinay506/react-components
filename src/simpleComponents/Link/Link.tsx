import React from 'react'
import styled from 'styled-components';
function Link(props: any) {
    /**
     * Text: The content which need to be displayed on the link
     * OnLinkClick is an output emitter when the user clicks on the link.
     */
    const { text, onLinkClick, color, fontSize, display } = props;

    const overrideStyles = {
        color: color,
        fontSize: fontSize,
        display: display
    }

    return (
        <React.Fragment>
            <Anchor id="anchor" onClick={onLinkClick} style={overrideStyles}>
                {text}
            </Anchor>
        </React.Fragment>
    )
}

const Anchor = styled.div`
     color: #0275d8;
    cursor: pointer;
    font-size: 16px;
    font-weight: 600;
    `;
export default Link
