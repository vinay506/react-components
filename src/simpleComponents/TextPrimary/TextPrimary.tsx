import React from 'react';
import styled from 'styled-components';
function TextPrimary(props: any) {
    /**
     * Text: The text which should be displayed on web page 
     * color: Defaults to black, To override the existing color we can pass Color code or color name as props.
     */

    return (
        <React.Fragment>
            <Label id="textPrimary" style={props.styles}>
                {props.text}
            </Label>
        </React.Fragment >
    )
}

const Label = styled.div`
    color: black;
    font-size: 21px;
    font-family: 'AmazonEmber-Bold';
     letter-spacing: 0;
  line-height: 28px;
    `;
export default TextPrimary;
