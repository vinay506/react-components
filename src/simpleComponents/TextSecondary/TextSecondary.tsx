import React from 'react'
import styled from 'styled-components';
function TextSecondary(props: any) {
    /**
     * Text: The text which should be displayed on web page
     * color: Defaults to black, To override the existing color we can pass Color code or color name as props.
     */
    const { text, styles } = props;
    
    return (
        <React.Fragment>
            <Label id="textSecondary" style={styles}>
                {text}
            </Label>
        </React.Fragment >
    )
}

const Label = styled.div`
    font-size: 18px;
    color: black;
    font-family: 'AmazonEmber-bold'
    `;

export default TextSecondary
