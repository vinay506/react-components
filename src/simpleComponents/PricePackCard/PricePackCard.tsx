import React from 'react'
import '@rmwc/card/styles';
import { Card } from '@rmwc/card';
import styled from 'styled-components';
import TextSecondary from '../TextSecondary/TextSecondary';
import TextContent from '../TextContent/TextContent';
import Link from '../Link/Link';
import CustomButton, { BUTTON_TYPES } from '../CustomButton/CustomButton';
import TextPrimary from '../TextPrimary/TextPrimary';
import CustomSymbol from '../CustomSymbol/CustomSymbol';
import CustomBadge from '../CustomBadge/CustomBadge';
function PricePackCard(props: any) {
    /**
     * badgeText: Text on the badge if not passed it will not be rendered on the card
     * logo: image URL, If not passed it will be ignored from the card
     * titleText: Primary text to show on the card 
     * packText: Secondary text to show on the card
     * scheduleLink: The link label to show on the card
     * rupees: Cost to show on the card
     * expiryYearText: Text to show with continuation of cost
     * buttonText: Text to show on the button
     * buttonCallBack: Call back method when user clicks on the button
     */
    const { badgeText, logo, titleText, packText, targetText, scheduleLink, rupees, expiryYearText, buttonText, buttonCallBack, linkCallBack } = props;
    const defaultStyles = {
        width: '250px',
        height: '380px',
        border: '1px solid lightgray',
        padding: '10px'
    }
    const showBadge = badgeText ? <CustomBadge text={badgeText} bgColor={'darkgray'} color={'white'} position={'absolute'} top={'-10px'} left={'15px'}></CustomBadge> : <></>;
    const showLogo = logo ? <img height="50" width="50" src={logo} /> : <div style={{ height: '50px', width: '50px' }}></div>
    const showTitle = titleText ? <TextSecondary text={titleText}></TextSecondary> : <></>;
    const showPackage = packText ? <TextContent text={packText}></TextContent> : '';
    const showTargetText = targetText ? <TextContent text={targetText}></TextContent> : <></>;
    const showLink = scheduleLink ? <Link text={scheduleLink} onLinkClick={linkCallBack}></Link> : <></>;
    const showRupees = rupees ? <TextPrimary text={rupees}></TextPrimary> : <></>;
    const showExpiry = expiryYearText ? <TextContent text={expiryYearText}></TextContent> : <></>;
    const showButton = buttonText ? <CustomButton text={buttonText} type={BUTTON_TYPES.OUTLINED} bgColor={'white'} borderColor={'#0275d8'} color={'#0275d8'} onButtonClick={buttonCallBack}></CustomButton> : <></>;
    return (
        <React.Fragment>
            <Card style={defaultStyles} id="uniqueCard">
                {showBadge}
                <ImageContainer>
                    {showLogo}
                </ImageContainer>
                <SecondaryContainer>
                    {showTitle}
                </SecondaryContainer>
                <PackContainer>
                    {showPackage}
                </PackContainer>
                <PackContainer>
                    {showTargetText}
                </PackContainer>
                <LinkContainer>
                    {showLink}
                </LinkContainer>
                <HorizantalLine></HorizantalLine>
                <AmountContainer>
                    <span>
                        <CustomSymbol color={'black'} fontSize={'24px'}>
                            &#8377;
                        </CustomSymbol>
                    </span>
                    <span>
                        {showRupees}
                    </span>
                    <CustomSymbol color={'black'}>
                        &nbsp;
                        </CustomSymbol>
                    <YearContainer>
                        {showExpiry}
                    </YearContainer>
                </AmountContainer>
                <ButtonContainer>
                    {showButton}
                </ButtonContainer>
            </Card>
        </React.Fragment>
    )
}


const SecondaryContainer = styled.div`
 height: 60px
`

const ButtonContainer = styled.div`
`
const YearContainer = styled.div`
padding-bottom: 0px;
`
const AmountContainer = styled.div`
display: flex;
justify-content: center;
margin-bottom: 10px;
`
const PackContainer = styled.div`
height: 30px;
`
const HorizantalLine = styled.div`
border: 0.8px solid lightgray;
margin-bottom: 20px;
`
const ImageContainer = styled.div`
    display: 'flex';
    justify-content: 'center';
    margin-top: 10px;
`

const LinkContainer = styled.div`
height: 40px;
`
export default PricePackCard
