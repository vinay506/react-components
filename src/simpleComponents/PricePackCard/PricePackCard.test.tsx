import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import PricePackCard from './PricePackCard';
import CustomBadge from '../CustomBadge/CustomBadge';
import TextSecondary from '../TextSecondary/TextSecondary';
import TextPrimary from '../TextPrimary/TextPrimary';
import CustomButton from '../CustomButton/CustomButton';
import Link from '../Link/Link';
import { TextContent } from '..';
describe('Price Pack Testing', () => {
    let wrapper: any;
    let propInfo: any;
    let defaultStyles: any;
    const callBack = () => {
        return true
    }
    const linkCallBack = () => {
        return true;
    }
    beforeEach(() => {
        propInfo = {
            badgeText: 'Recommended',
            titleText: 'JEE',
            packText: '1 year pack',
            targetText: 'Target Year 2020',
            scheduleLink: 'View Schedule',
            rupees: '3000',
            expiryYearText: 'for 1 year',
            buttonText: 'Get Started',
            buttonCallBack: callBack,
            linkCallBack: linkCallBack
        };
        defaultStyles = {
            width: '250px',
            height: '380px',
            border: '1px solid lightgray',
            padding: '10px'
        }
        wrapper = shallow(<PricePackCard {...propInfo} />);
    });
    test('Price Pack Card Snapshot', () => {
        const instance = renderer.create(<PricePackCard {...propInfo} />);
        const tree = instance.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Price Pack Card Existance', () => {
        // const instance = renderer.create(<PricePackCard {...propInfo} />);
        const elementLength = wrapper.find('#uniqueCard').length;
        expect(elementLength).toEqual(1);
    })
    test('Price Pack Badge Existance', () => {
        const count = wrapper.find(CustomBadge).length;
        if (propInfo.badgeText) {
            expect(count).toEqual(1);
        } else {
            expect(count).toEqual(0);
        }
    })

    test('Price Pack Title Existance', () => {
        const titleCount = wrapper.find(TextSecondary).length;
        if (propInfo.titleText) {
            expect(titleCount).toEqual(1);
        } else {
            expect(titleCount).toEqual(0);
        }
    })


    test('Price Pack Rupees Existance', () => {
        const count = wrapper.find(TextPrimary).length;
        if (propInfo.rupees) {
            expect(count).toEqual(1);
        } else {
            expect(count).toEqual(0);
        }
    })

    test('Price Pack Custom Button Existance', () => {
        const count = wrapper.find(CustomButton).length;
        if (propInfo.buttonText) {
            expect(count).toEqual(1);
        } else {
            expect(count).toEqual(0);
        }
    })

    test('Price Pack packText Existance', () => {
        const textContentArr = wrapper.find(TextContent);
        if (propInfo.packText) {
            let isExist = false;
            textContentArr.forEach((obj: any) => {
                if (obj && obj.props()) {
                    if (obj.props().text === propInfo.packText) {
                        isExist = true;
                    }
                }
            })
            expect(isExist).toBeTruthy();
        }
    })

    test('Price Pack targetText Existance', () => {
        const textContentArr = wrapper.find(TextContent);
        if (propInfo.targetText) {
            let isExist = false;
            textContentArr.forEach((obj: any) => {
                if (obj && obj.props()) {
                    if (obj.props().text === propInfo.targetText) {
                        isExist = true;
                    }
                }
            })
            expect(isExist).toBeTruthy();
        }
    })

    test('Price Pack expiryYearText Existance', () => {
        const textContentArr = wrapper.find(TextContent);
        if (propInfo.targetText) {
            let isExist = false;
            textContentArr.forEach((obj: any) => {
                if (obj && obj.props()) {
                    if (obj.props().text === propInfo.expiryYearText) {
                        isExist = true;
                    }
                }
            })
            expect(isExist).toBeTruthy();
        }
    })

    test('Price Pack Custom Link Existance', () => {
        const count = wrapper.find(Link).length;
        if (propInfo.scheduleLink) {
            expect(count).toEqual(1);
        } else {
            expect(count).toEqual(0);
        }
    })

    test('Price Pack Button Click Test', () => {

        const button = wrapper.find(CustomButton);
        const clickEvent = button.simulate('click');
        const returnValue = expect(clickEvent).toBeCalled;
        expect(returnValue).toBeTruthy();
    })

    test('Price Pack Link Click Test', () => {

        const link = wrapper.find(Link);
        const clickEvent = link.simulate('click');
        const returnValue = expect(clickEvent).toBeCalled;
        expect(returnValue).toBeTruthy();
    })

    // Default Styles for Card


    test('Price Pack width Validation', () => {
        const element = wrapper.find('#uniqueCard');
        expect(element.props().style.width).toEqual(defaultStyles.width);
    })

    test('Price Pack height Validation', () => {
        const element = wrapper.find('#uniqueCard');
        expect(element.props().style.height).toEqual(defaultStyles.height);
    })

    test('Price Pack border Validation', () => {
        const element = wrapper.find('#uniqueCard');
        expect(element.props().style.border).toEqual(defaultStyles.border);
    })

    test('Price Pack padding Validation', () => {
        const element = wrapper.find('#uniqueCard');
        expect(element.props().style.padding).toEqual(defaultStyles.padding);
    })
})

