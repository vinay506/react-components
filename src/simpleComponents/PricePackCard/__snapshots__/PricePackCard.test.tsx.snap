// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Pack Testing Price Pack Card Snapshot 1`] = `
<div
  className="mdc-card"
  id="uniqueCard"
  style={
    Object {
      "border": "1px solid lightgray",
      "height": "380px",
      "padding": "10px",
      "width": "250px",
    }
  }
>
  <span
    className="sc-AxhUy kvjJUJ"
    id="uniqueBadge"
    style={
      Object {
        "backgroundColor": "darkgray",
        "borderRadius": undefined,
        "bottom": undefined,
        "color": "white",
        "left": "15px",
        "position": "absolute",
        "right": undefined,
        "top": "-10px",
      }
    }
  >
    Recommended
  </span>
  <div
    className="sc-fzpans bJETly"
  >
    <div
      style={
        Object {
          "height": "50px",
          "width": "50px",
        }
      }
    />
  </div>
  <div
    className="sc-AxgMl kALZGm"
  >
    <div
      className="sc-AxjAm hLZUJk"
      id="textSecondary"
    >
      JEE
    </div>
  </div>
  <div
    className="sc-fzozJi lbrNyh"
  >
    <div
      className="sc-AxirZ ixsMyO"
      id="textContent"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      1 year pack
    </div>
  </div>
  <div
    className="sc-fzozJi lbrNyh"
  >
    <div
      className="sc-AxirZ ixsMyO"
      id="textContent"
      style={
        Object {
          "color": undefined,
        }
      }
    >
      Target Year 2020
    </div>
  </div>
  <div
    className="sc-fzplWN ksGGeZ"
  >
    <div
      className="sc-AxiKw hukwlf"
      id="anchor"
      onClick={[Function]}
      style={
        Object {
          "color": undefined,
          "display": undefined,
          "fontSize": undefined,
        }
      }
    >
      View Schedule
    </div>
  </div>
  <div
    className="sc-fzoLsD jKdIiq"
  />
  <div
    className="sc-AxmLO jktqSx"
  >
    <span>
      <div
        style={
          Object {
            "color": "black",
            "fontSize": "24px",
          }
        }
      >
        ₹
      </div>
    </span>
    <span>
      <div
        className="sc-AxhCb ekSghy"
        id="textPrimary"
      >
        3000
      </div>
    </span>
    <div
      style={
        Object {
          "color": "black",
          "fontSize": undefined,
        }
      }
    >
       
    </div>
    <div
      className="sc-Axmtr iWYdZe"
    >
      <div
        className="sc-AxirZ ixsMyO"
        id="textContent"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        for 1 year
      </div>
    </div>
  </div>
  <div
    className="sc-AxheI dVkpdt"
  >
    <button
      className="mdc-button mdc-button--outlined"
      id="outlined"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#0275d8",
          "color": "#0275d8",
        }
      }
    >
      <div
        className="mdc-button__ripple"
      />
      <span
        className="mdc-button__label"
      >
        Get Started
      </span>
    </button>
  </div>
</div>
`;
